Godot XR Update Sep 2023

BastiaanOlij opened the pull request:
Update for XR work on Godot, put together by the Godot XR community.

Just needs some cover art but otherwise done.

Would love more feedback

BastiaanOlij committed:
Godot XR Update Sep 2023

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+## New plugin structure on Android
+
+Thanks to the work of Fredia Huya-Kouadio Godots Android support for the upcoming Godot 4.2 release has had a big refactor of its plugin structure. This means that the Godot OpenXR loaders plugin has been streamlined and is now able to support newer OpenXR extensions that require platform specific feature tags and permissions.
```
```suggestion
Thanks to the work of Fredia Huya-Kouadio, Godot's Android support for the upcoming Godot 4.2 release has had a big refactor of its plugin structure. This means that the Godot OpenXR loaders plugin has been streamlined and is now able to support newer OpenXR extensions that require platform specific feature tags and permissions.
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+OpenXR regularly sees new vendor extensions being added for functionality that is only available on a single platform or where there are other reasons that prevent adding support to the core of Godot.
+
+Migeran have upstreamed their improvements to Godot to allow these extensions to be implemented through Godots GDExtension system.
```
```suggestion
Migeran have upstreamed their improvements to Godot to allow these extensions to be implemented through Godot's GDExtension system.
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+Migeran have upstreamed their improvements to Godot to allow these extensions to be implemented through Godots GDExtension system.
+
+Currently they are working on a plugin that implements Metas set of scene and spatial extensions.
```
```suggestion
Currently they are working on a plugin that implements Meta's set of scene and spatial extensions.
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+Currently they are working on a plugin that implements Metas set of scene and spatial extensions.
+
+## OpenXR MacOS Support
```
```suggestion
## OpenXR macOS Support
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+## OpenXR MacOS Support
+
+OpenXR added MacOS support to the latest specification. While XR runtimes are still under development Godot now supports this functionality.
```
```suggestion
OpenXR added macOS support to the latest specification. While XR runtimes are still under development Godot now supports this functionality.
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+Having game engine representation in the working group is of utmost importance in ensuring the standard caters to the needs of XR game developers.
+
+## Godot TiltFive Support
```
```suggestion
## Godot Tilt Five Support
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+
+## Status update Godot XR Tools
+
+Our XR Tools library has seen a steady flow of improvements over the past months and has some really interesting improvements in the pipeline. We release version 4.2.0 on the 22nd of August with a long list of improvements.
```
```suggestion
Our XR Tools library has seen a steady flow of improvements over the past months and has some really interesting improvements in the pipeline. We released version 4.2.0 on the 22nd of August with a long list of improvements.
```

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,81 @@
+Thanks to contributor DigitalN8m4r3 the demo has undergone a nice visual overhaul.
+<add image>
+
+He has also contributed a new audio system to XR Tools that makes it easy to create audio que’s as a player moves through the virtual space and interacts with various objects.
```
```suggestion
He has also contributed a new audio system to XR Tools that makes it easy to create audio cues as a player moves through the virtual space and interacts with various objects.
```

BastiaanOlij commented:
Thanks for checking @JohnVeness !

BastiaanOlij commented:
Added small section about PICO as well, with their new OS release earlier this month the Vulkan renderer is finally working properly and they also provided a device for us to test with.

Similarly also added mention of the test unit Tilt Five supplied to us.

This way Metas, Valves, Picos and Tilt Fives contributions are all noted.

clayjohn commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,88 @@
+
+There is still plenty happening on the XR front to warrant another update.
+
+We again thank Meta for the support past and present that they have been able to provide.
```
I would shift the focus away from Meta in these first couple of paragraphs. There is no reason to thank them for a grant that didn't fund the described work. Similarly, we want to highlight both that we need an XR grant to shift the focus back to XR, and that the grant can come from anyone (It would be great if it came from Meta as we have worked well with them in the past, but other companies are free to step up as well).

I would go with something a little bit subtler
```suggestion
XR work has slowed down this year due to the XR-specific line of funding running out. However, there is still plenty happening on the XR front to warrant another update.
```

BastiaanOlij replied:
I didn't want to come across too harsh, but yeah us currently not having any funding for XR has had a measurable impact which is why I wanted to highlight that. I like your rewording

JohnVeness commented on collections/_article/godot-xr-update-aug-2023.md:
```diff
@@ -0,0 +1,88 @@
+
+PICO has been working closely with the Godot XR Team to ensure Godot runs properly on PICO devices as well. This included making a PICO 4 headset available to the team.
+
+With the latest OS update on the PICO headset both compatibility and Vulkan mobile renderers are working well on the device.
```
```suggestion
With the latest OS update on the PICO headset both Compatibility and Vulkan mobile renderers are working well on the device.
```
So it's clearer that compatibility is a renderer.

BastiaanOlij commented:
Ok, images are added. I don't have a build setup so not sure if there is an easy way to verify everything renders properly?

coppolaemilio commented:
This is how it looks like to me @BastiaanOlij 
It would be nice to have another image or if possible a small video? It is always much nicer to see a short clip of a vr demo than a still image
![blogpost](https://github.com/godotengine/godot-website/assets/2206700/74697eb8-addd-4d2a-a92f-88a05868c112)

BastiaanOlij commented:
@coppolaemilio not sure what the format should be but that image from Digital comes from one of the videos he posted on his PR: https://github.com/GodotVR/godot-xr-tools/pull/478
We could replace the still with the video, it looks really nice.

Also I could ask Nub if he's happy to sent us on of his videos, his game is coming along nicely too: https://twitter.com/nubusvr/status/1679478903520239617?s=20

BastiaanOlij commented:
@coppolaemilio I've added in the video from Digital instead of the still image, I hope the file isn't too big to include on our website (do we have restrictions here?).

I've also added a link to a video Nub uploaded on his Youtube for me showing off his gunslinger game he is working on. Thats a pretty neat project.

I think this is pretty much ready to go, would be good if you could give it another visual check?

coppolaemilio committed:
update release date and thumbnail

coppolaemilio commented:
Thank you!
